src/sync.c

```diff
@@ -91,7 +94,9 @@ wait_barrier(barrier_t *bar)
 	assert(bar);
 	assert(bar->limit > 0);
 
-#ifdef HAVE_ATOMIC_H
+#ifdef HAVE_STDATOMIC_H
+	(void) atomic_fetch_add(&bar->count, 1);
```

Comment:
Is `atomic_fetch_add` exactly equivalent to `atomic_add_32_nv`?  I ask because `atomic_fetch_add` is also replacing `atomic_add_32` (no `_nv`) in `src/shm.c`.

Reply:
And it looks [1] like `atomic_fetch_add` works on 32 bit values, while `atomic64_fetch_add` and `atomic_long_fetch_add` variants work on 64 bit values, so the sizing looks correct.

[1] https://www.kernel.org/doc/html/latest/staging/index.html#atomic-types